### PR TITLE
Separated test engine configuration from the actual application JSON

### PR DESCRIPTION
**Pull request by apandurangi**
feat(server): Added support to engine configuration file for local engine.

  - Added a new app config property `defaultFlogoEngineConfigPath`
  - Users can set environment variable FLOGO_WEB_DEFAULT_ENGINE_CONFIG to set the path to default engine config
  - Users can provide a different path while updating the engine config

## Description

The latest flogo core provides support to maintain different engine configuration files which will be allow us to build the flogo-web's local engine binary with different test configurations.

## How has this been tested?

Manual testing. 
1. Should not effect the new engine creation while the server start up
2. The telemetry service must be installed to the local engine through this engine config file
3. Should maintain a `engine.json` file in local engine's folder before building the local engine

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build/CI related changes
- [ ] Documentation related changes
- [ ] Other... Please describe:

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/server/src/modules/engine/registry.ts [hunk 1]
```diff
@@ -86,6 +92,9 @@ async function createEngine(engine, defaultFlogoDescriptorPath, skipBundleInstal
     await installResourceTypes(engine, defaultResourceTypes);
     await engine.installContribBundle(contribBundlePath);
     await tempInstallSimulatorDeps(engine.getProjectDetails());
+    if (bundleEngineConfig) {
```
Comment (apandurangi) [inline apps/server/src/modules/engine/registry.ts]: @fcastill I just realized, this should be moved to the top immediately after creating the engine. As bundling the config file and installing the contribution from bundle file are not related to each other. So setting the `skipBundleInstall` to false should **not** stop bundling the engine configuration. What do you think?

Comment (fcastill) [reply apps/server/src/modules/engine/registry.ts]: true, I didn't see that, same for install resource types, I guess

Comment (fcastill) [reply apps/server/src/modules/engine/registry.ts]: I mean something like:
```
if (!skipBundleInstall) {
  // await install bundle
}
await installResourceTypes();
if (bundleEngineConfig) {
  // await update engine config
}
```

which makes me wish the options were `skipContribBundle`/`skipEngineConfig` or `useContribBundle`/`useEngineConfig`

Comment (apandurangi) [reply apps/server/src/modules/engine/registry.ts]: `useContribBundle` `useEngineConfig` sounds good !! 

I think skipping `installResourceType` is apt there as while generating a binary from UI we do not want to install all resource types. I think we can club both resource type installation and contribution bundle installation with `useContribBundle` flag.

I will implement following for now: 
1. change the names of the flags
2. instead of returning out of the function I will do:
```
if (useContribBundle) {
  // await install bundle
await installResourceTypes();
}
if (useEngineConfig) {
  // await update engine config
}
```
